src/Sparrow/Threading/DisposeOnce.cs

```diff
@@ -7,6 +7,7 @@ namespace Sparrow.Threading
     public interface IDisposeOnceOperationMode {}
     public struct ExceptionRetry : IDisposeOnceOperationMode { }
     public struct SingleAttempt : IDisposeOnceOperationMode { }
+    public struct SingleAttemptWithWaitForDisposeToFinish : IDisposeOnceOperationMode { }
```

Comment:
I don't like this, we should add a `disposing` flag for when we run the actual disposing

Reply:
done